### PR TITLE
Prepare to use workspaces for tests using linchpin fetch

### LINKED ISSUE #618: RFE: `--no-cache` option for fetch
`fetch` command by default uses cache `~/.cache/linchpin/...`, there is no option to clean cache or not to use it at all. Expected result:
```
linchpin fetch --no-cache FETCH_TYPE REMOTE
```

### LINKED ISSUE #619: RFE: fetch overwrite option
By default, fetch doesn't overwrite files, option to do so can be helpful for resetting the environment. For example: `linchpin fetch --force FETCH_TYPE REMOTE`

### PR DESCRIPTION
**Pull request by herlo**
Fixes #618 
Fixes #619 

The linchpin repository will have multiple workspaces in the examples section (with a symlink from the root 'workspaces' path).

In the future, tests, examples, etc. will use workspaces. There will be a README.rst and a standard set of items to add to each workspace as well. The workspaces now exist.

Some work on the linchpin fetch functionality needed to occur as well to ensure pulling of a particular branch or reference.

### CODE REVIEW

**Comment by herlo** [comment]
[test]

File: docs/source/conf.py [hunk 1]
```diff
@@ -357,4 +357,9 @@
     'https://raw.githubusercontent.com/CentOS-PaaS-SIG/linchpin/develop/%s', ''),
     'dirs1.5': (
     'https://github.com/CentOS-PaaS-SIG/linchpin/tree/develop/docs/source/examples/%s', ''),
+    'lp_dir': (
+    'https://github.com/herlo/linchpin/tree/fetch_by_branch/%s', ''),
+    'lp_code': (
+    'https://raw.githubusercontent.com/herlo/linchpin/fetch_by_branch/%s', ''),
```
[the remark this answers is not in the document]
Comment (herlo) [reply docs/source/conf.py]: @14rcole yes, it should be, but until this is merged, the docs don't point to the right place. I'll update this before merging.

File: docs/source/examples/workspaces/dummy-aws/PinFile.dummy-aws.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+---
```
Comment (14rcole) [inline docs/source/examples/workspaces/dummy-aws/PinFile.dummy-aws.yml]: why is dummy-aws in its own workspace while others like dummy-openstack are in random?

Comment (herlo) [reply docs/source/examples/workspaces/dummy-aws/PinFile.dummy-aws.yml]: good point. I'll fix that up.

**Comment by herlo** [comment]
[test]

**Comment by herlo** [comment]
[test]

File: docs/source/workflow.rst [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+As expected, the ``simple`` root was pulled down, and placed inside the
+``/tmp/workspaces`` directory on the local machine.
+
+To have **all** workspaces copied into /tmp/workspaces, a change is needed.
+
+.. code-block:: bash
+
+    $ linchpin fetch git://github.com/CentOS-PaaS-SIG/linchpin \
+    > --root workspaces --dest /tmp
```
Comment (samvarankashyap) [inline docs/source/workflow.rst]: @herlo r u sure the --root workspaces is correct ? isnt  docs/source/examples/workspaces
Will linchpin search all the sub  directories of a git repository ?

Comment (samvarankashyap) [reply docs/source/workflow.rst]: @herlo or does the --root workspaces defaults to docs/source/examples/workspaces ?

Comment (herlo) [reply docs/source/workflow.rst]: @samvarankashyap if you look at the root of the linchpin directory in this PR, you'll see a symlink from workspaces to docs/source/examples/workspaces.

Comment (herlo) [reply docs/source/workflow.rst]: `linchpin fetch` handles symlinks, and this helps prove that out.

**Comment by herlo** [comment]
[merge]

**Pull request merged**